frontend/views/containers/payments/PaymentsList.vue

```diff
@@ -44,6 +59,68 @@ export default ({
       validator: (value) => ['PaymentRowTodo', 'PaymentRowSent', 'PaymentRowReceived'].includes(value),
       required: true
     }
+  },
+  data () {
+    return {
+      form: {
+        checkAll: false
+      },
+      config: {
+        debouncedOnItemChange: debounce(this.onItemChange, 100)
+      }
+    }
+  },
+  computed: {
+    tableClass () {
+      const map = {
+        'PaymentRowTodo': 'todo',
+        'PaymentRowSent': 'sent',
+        'PaymentRowReceived': 'received'
+      }
+
+      return `c-is-${map[this.paymentsType]}`
+    }
+  },
+  watch: {
+    'form.checkAll' (val) {
+      const method = val ? 'select' : 'deselect'
+
+      this.$refs.paymentItem.forEach(c => c[method](true))
+    },
+    paymentsType: {
+      immediate: true,
+      handler (val) {
+        if (val === 'PaymentRowTodo') {
+          // each time the tab in Payments.vue is switched back to 'PaymentsTodo',
+          // ephemeral.selectedTodoItems needs to get updated accordingly so that
+          // the 'Send payments' button updates its 'disabled' attribute properly.
+          const sendCurrentlySelectedTodos = () => {
+            this.$emit('todo-items-change', this.getAllSelectedTodoItems())
+          }
+
+          if (this.$refs.paymentItem) {
+            sendCurrentlySelectedTodos()
+          } else {
+            // if the children are not mounted yet, give it a bit of delay until then.
+            setTimeout(sendCurrentlySelectedTodos, 50)
+          }
+        }
+      }
+    }
+  },
```

Comment:
Woah, this looks weird... making a note to self to review this in more detail.

@pieer any thoughts on this?

Reply:
@taoeffect 
The reason for this was that `ephemeral.selectedTodoItems` in the parent component(`Payments.vue`) needs to be updated as `paymentsType` prop here changes back to `PaymentRowTodo`. I agree things here are written verbosely. so I have improved the logic and removed the `setTimeout` there in [this following commit](https://github.com/okTurtles/group-income/pull/1390/commits/df9098f0ce549f3b8c8510ac448116d069c9afa7#diff-c5b86c07ceca3303b10a31f22491d980e6f37476ee997b269268c913e5777c0f). hope things look better.